### PR TITLE
REP-166 Adding unittests

### PR DESCRIPTION
**Pull request by dovekie**
@alexanderyoung PTAL

### CODE REVIEW

File: bin/critterget.py [hunk 1]
```diff
@@ -121,6 +91,10 @@ def scopetime():
 
     return (newtime.isoformat())
 
+# consider replacing 'print' statements with optional 'out' parameters, e.g.:
```
Comment (alexanderyoung) [inline bin/critterget.py]: Do we need this comment in the code?

File: bin/critterget.py [hunk 2]
```diff
@@ -38,23 +41,23 @@ def apicall (uri, attribs=''):
...
 
-    except urllib2.URLError, e:
+    except requests.exceptions.RequestException as e:
         print 'Crittercism API retuned an error code:', e
         sys.exit(0)
 
     return data
 
-def apipost (uri, postdata='',keyget=''):
+def apipost (uri, postdata='', keyget=''):
```
Comment (alexanderyoung) [inline bin/critterget.py]: Whenever you're modifying a function like this clean it up a little if you can.

I would replace the default values for postdata and keyget with None here.

File: bin/critterget.py [hunk 3]
```diff
@@ -64,56 +67,23 @@ def apipost (uri, postdata='',keyget=''):
 
     pdata = json.dumps(postdata)
 
-    request = urllib2.Request(reqstring,pdata)
     if (keyget==''):
```
Comment (alexanderyoung) [inline bin/critterget.py]: This can just say:
if not keyget:

File: bin/critterget.py [hunk 4]
```diff
@@ -38,23 +41,23 @@ def apicall (uri, attribs=''):
...
         print 'Crittercism API retuned an error code:', e
         sys.exit(0)
 
     return data
 
-def apipost (uri, postdata='',keyget=''):
+def apipost (uri, postdata='', keyget=''):
     # perform an API POST
     if (debug) : print u'access token is {}'.format(access_token)
     reqstring = baseurl + uri
```
Comment (alexanderyoung) [inline bin/critterget.py]: I can't comment on the line below for some reason.  It seems like 'data' doesn't need to exist on line 64.

File: bin/critterget.py [hunk 5]
```diff
@@ -64,56 +67,23 @@ def apipost (uri, postdata='',keyget=''):
...
+            'Authorization': "Bearer %s" % access_token,
+            'CR-source': 'integration_splunk'
+        }
 
     try:
-        response = urllib2.urlopen(request)
-        resptext = response.read()
-        data = json.loads(resptext)
+        response = requests.post(reqstring, headers=headers, data=pdata)
+        data = response.json()
```
Comment (alexanderyoung) [inline bin/critterget.py]: This could just be:
return response.json()

File: bin/critterget.py [hunk 6]
```diff
@@ -64,56 +67,23 @@ def apipost (uri, postdata='',keyget=''):
...
 
-    except urllib2.URLError, e:
+    except requests.exceptions.RequestException as e:
         print u'{} MessageType="CrittercismError" Crittercism API retuned an error code: {} for the call to {}.  Maybe an ENTERPRISE feature?'.format(myruntime, e, reqstring)
         data = "ERROR"
```
Comment (alexanderyoung) [inline bin/critterget.py]: This could just be return 'ERROR'

File: bin/critterget.py [hunk 7]
```diff
@@ -1,15 +1,18 @@
...
 import json
 import sys
-import datetime, base64
-import splunk.entity as entity
+import requests
+
+try:
+    import splunk.entity as entity
+except ImportError:
+    import splunk_mock.entity as entity
```
Comment (alexanderyoung) [inline bin/critterget.py]: No way to handle this in your unittest code?

**Comment by alexanderyoung** [comment]
Generally looks very good, please take a look at the comments I left though.

Good work.

File: bin/critterget.py (hunk 7 above)
Comment (dovekie) [inline bin/critterget.py]: This seemed like the simplest solution, and mimics patterns I've seen elsewhere in our code base.

**Comment by alexanderyoung** [comment]
LGTM

**Pull request merged**